nix/packages.nix

```diff
@@ -21,81 +21,159 @@ inputs.nixpkgs.lib.genAttrs systems
         hash = "sha256-FLPr0AbSyPDtl4CLJus4l9wCmXEQlCHLStuTw5K1hFI=";
       };
 
-      targets = [
-        "linux/amd64"
-        "linux/arm64"
-        "darwin/arm64"
-      ];
+      goSumSha = "sha256-mrHfOS9EoM8o9RXQBYVLGFR+uGITI3v/aRAKBq3/wc0=";
+
+      commonCheckPhase = ''
+        # point npm at the offline cache
+        export npm_config_cache=${npmDeps}
+
+        # React lint
+        cd react-display
+        npm ci --cache="$npm_config_cache" --prefer-offline --no-audit --progress=false
+        node node_modules/.bin/eslint . --max-warnings=0
+        cd ..
+
+        # Go tests
+        export CGO_ENABLED=1
+        export XDG_CACHE_HOME=$TMPDIR
+        mkdir -p $XDG_CACHE_HOME/staticcheck
+        staticcheck ./...
+        go test -count=1 --race -v ./...
+      '';
+
+      reactBuild = ''
+        # point npm at the offline cache
+        export npm_config_cache=${npmDeps}
+
+        # build the React bundle
+        cd react-display
+        npm ci --cache="$npm_config_cache" --prefer-offline --no-audit --progress=false
+
+        # invoke tsc & vite via node
+        node node_modules/.bin/tsc -b
+        node node_modules/.bin/vite build
+        cd ..
+      '';
 
     in
     {
+      scale-simulator = pkgs.buildGoModule rec {
```

Comment:
this is going to be really cool to have running on the demo site!

You can replace `pkgs.buildGoModule rec {` with `buildGoModule (finalAttrs: {` like:
https://github.com/NixOS/nixpkgs/blob/f771eb401a46846c1aebd20552521b233dd7e18b/pkgs/by-name/op/opkssh/package.nix#L9 This makes it so pkg overriding is manageable.  I was trying to find the gh issue since that was something thats happened in the last couple months or so.